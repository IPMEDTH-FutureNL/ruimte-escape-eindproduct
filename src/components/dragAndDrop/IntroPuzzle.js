import React, { useState } from 'react'
import { useNavigate } from "react-router-dom";

import PopUp from '../PopUp';

const IntroPuzzle = () => {
    const [fact, setFact] = useState("");
    const navigate = useNavigate();

    const startPuzzel = () => {
        navigate("/introPuzzelEen");
    }

    const feitjes = (feit) => {
        if(feit === 1){
            setFact("Wist je dat Mercurius de kleinste planeet is!");
            let openpopup = document.getElementById("popup");
            openpopup.style.display = "block";
        }else if(feit === 2){
            setFact("De eerste mens op de maan was Neil Armstrong!");
            let openpopup = document.getElementById("popup");
            openpopup.style.display = "block";
        }else{

        }
    }

    return(
        <section className="container__intro-startpuzzel">
            <div className="circle" onClick={startPuzzel}></div>
            <div className="circle" onClick={() => feitjes(1)}></div>
            <div className="circle" onClick={() => feitjes(2)}></div>

            <PopUp text={fact}></PopUp>
        </section>
    )
}

export default IntroPuzzle;