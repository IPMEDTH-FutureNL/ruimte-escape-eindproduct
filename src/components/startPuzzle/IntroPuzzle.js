import React, { Component } from 'react'
import { useNavigate } from "react-router-dom";

const IntroPuzzle = () => {

    const navigate = useNavigate();

    const startPuzzel = () => {
        navigate("/startpuzzel");
    }

    const feitjes = (feit) => {
        if(feit === 1){
            alert("Wist je dat Mercurius de kleinste planeet is!");
        }else if(feit === 2){
            alert("De eerste mens op de maan is Neil Armstrong!");
        }else{

        }
    }

    return(
        <section className="container__intro-startpuzzel">
            <div className="circle" onClick={startPuzzel}></div>
            <div className="circle" onClick={() => feitjes(1)}></div>
            <div className="circle" onClick={() => feitjes(2)}></div>
        </section>
    )
}

export default IntroPuzzle;