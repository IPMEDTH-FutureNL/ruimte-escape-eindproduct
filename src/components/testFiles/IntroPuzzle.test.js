import renderer from 'react-test-renderer';
import IntroPuzzle from "../dragAndDrop/IntroPuzzle";
import {BrowserRouter as Router} from 'react-router-dom';

it('Test if IntroPuzzle component is rendered', () => {
    const component = renderer.create(
        <Router>
            <IntroPuzzle></IntroPuzzle>,
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})