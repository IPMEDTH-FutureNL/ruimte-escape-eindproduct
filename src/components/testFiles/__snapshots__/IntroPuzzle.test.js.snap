// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if IntroPuzzle component is rendered 1`] = `
Array [
  <section
    className="container__intro-startpuzzel"
  >
    <div
      className="circle"
      onClick={[Function]}
    />
    <div
      className="circle"
      onClick={[Function]}
    />
    <div
      className="circle"
      onClick={[Function]}
    />
    <div
      className="pop-up"
      id="popup"
    >
      <p
        className="pop-up-text"
      >
         
         
      </p>
      <button
        className="button primary"
        onClick={[Function]}
      >
        Verder!
      </button>
    </div>
  </section>,
  ",",
]
`;
